### PR TITLE
Add auto-generated Lua spec files for APIs, models

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Add auto-generated Lua spec files for APIs, models

cc @daurnimator

### CODE REVIEW

File: samples/client/petstore/lua/petstore/spec/category_spec.lua [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+This is a sample server Petstore server.  You can find out more about Swagger at [http://swagger.io](http://swagger.io) or on [irc.freenode.net, #swagger](http://swagger.io/irc/).  For this sample, you can use the api key `special-key` to test the authorization filters.
+
+OpenAPI spec version: 1.0.0
+Contact: apiteam@swagger.io
+Generated by: https://github.com/swagger-api/swagger-codegen.git
+Swagger Codegen version: 2.4.0-SNAPSHOT
+
+]]
+
+local petstore_category = require "petstore.model.category"
```
Comment (daurnimator) [inline samples/client/petstore/lua/petstore/spec/category_spec.lua]: Usually I put this `require` inside of the `describe`. That way if you're only running some tests or mocking out something, it won't require it too early.

File: samples/client/petstore/lua/petstore/spec/category_spec.lua [hunk 2]
```diff
@@ -0,0 +1,35 @@
+--[[
+Swagger Petstore
+
+This is a sample server Petstore server.  You can find out more about Swagger at [http://swagger.io](http://swagger.io) or on [irc.freenode.net, #swagger](http://swagger.io/irc/).  For this sample, you can use the api key `special-key` to test the authorization filters.
+
+OpenAPI spec version: 1.0.0
+Contact: apiteam@swagger.io
+Generated by: https://github.com/swagger-api/swagger-codegen.git
+Swagger Codegen version: 2.4.0-SNAPSHOT
+
```
Comment (daurnimator) [inline samples/client/petstore/lua/petstore/spec/category_spec.lua]: Extra newline?

File: samples/client/petstore/lua/petstore/spec/category_spec.lua [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    end)
+  end)
+
+  -- unit tests for the property 'name'
+  describe("property name test", function()
+    it("should work", function()
+      -- TODO assertion here: http://olivinelabs.com/busted/#asserts
+    end)
+  end)
+
```
Comment (daurnimator) [inline samples/client/petstore/lua/petstore/spec/category_spec.lua]: Extra newline

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/LuaClientCodegen.java [hunk 4]
```diff
@@ -270,21 +281,16 @@ public String toApiFilename(String name) {
...
-
-        //if (Character.isUpperCase(firstChar)) {
-        //    // First char is already uppercase, just use paramName.
-        //    parameter.vendorExtensions.put("x-exportParamName", parameter.paramName);
+    }
 
-        //}
+    @Override
+    public String apiTestFileFolder() {
+        return outputFolder + File.separator + apiPackage + File.separator + specFolder.replace("/", File.separator);
```
Comment (daurnimator) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/LuaClientCodegen.java]: The spec folder should be at the top level (same level as e.g. the rockspec)

**Comment by wing328** [comment]
@daurnimator thanks for the feedback. I'll address it over the weekend.

**Pull request merged**